WreckerRequest Interceptor [#4]

benpate opened the pull request:
Here are the base changes to intercept wrecker.Request objects only.
Will make additional PRs for HTTPRequest and HTTPResponse interceptors.

benpate committed:
WreckerRequest Interceptor [#4]

Here are the base changes to intercept wrecker.Request objects only.
Will make additional PRs for HTTPRequest and HTTPResponse interceptors.

BrandonRomano commented:
Hey @benpate can you look into why this travis build failed?

Here's the full log:

https://s3.amazonaws.com/archive.travis-ci.org/jobs/70090742/log.txt

benpate commented:
With a quick check of the logs, I'm not sure.  All tests are passing for me locally, so my first guess is to blame Travis.  But, I'll dig in more before I pass the buck.

BrandonRomano commented:
It likely is something that's only happening on travis' side.. Perhaps is a config issue?

I'll also give this a look too later.

benpate commented:
Yup.  Check out this line in the logs:

```
go get -t -v ./...
github.com/brandonromano/wrecker (download)
github.com/julienschmidt/httprouter (download)
github.com/stretchr/testify (download)
github.com/BrandonRomano/wrecker
github.com/brandonromano/wrecker/test/models
github.com/julienschmidt/httprouter
github.com/BrandonRomano/wrecker/test/models
github.com/BrandonRomano/wrecker/test
```

I added a new file called `interceptor.go` that includes the definition for the Interceptor object.  It looks like go get is not finding it for some reason.

I am noticing the difference in capitalization "github.com/BrandonRomano" vs "github.com/brandonromano" in several of the imports.  Do you think that has anything to do with it?

BrandonRomano commented:
That was definitely one of my hypothesis for why that was happening, but I think I remember testing it and it wasn't the case.  I was trying to figure this one out for like 2-3 hours a few nights ago (this is actually the same issue occurring as we were getting before the PR split up).

I think the best way to test this is to perhaps start a new branch + start adding some of the code slowly until we reach an error... Then we can at least see what lines are causing the issue.

benpate commented:
OK.  I'm going to put my new stuff into the existing files -- to see if my hunch is correct.  If it is, then I may need to ask you to look in to Travis more (I have -0 Travis experience).  In the long run, I think you'll definitely want to be able to add more files to the project without it breaking :)

benpate committed:
Relocate Interceptor definition

Moving type definition into the main wrecker.go file to test if this
helps the Travis tests to pass.

benpate committed:
Revert "Relocate Interceptor definition"

This reverts commit 83ebfa18bb44493e2a1edf9ab4f5d9dd32888cff.

benpate commented:
So it looks like this is still not working.  Unfortunately, it's above my pay grade.  I do know someone with some Hudson experience -- if you'd like to call in someone from the outside.  He'd probably appreciate the opportunity to work with the hosted solution.

BrandonRomano commented:
Hmm -- I'd prefer to keep it travis if at all possible... Perhaps I can dig into it as well.  I notice you've got another branch going and it seems like it's coming from the tests... Perhaps one of us can dig into that deeper.

benpate commented:
Oh, sorry.  I wasn’t suggesting that we switch off of Travis.  I thought that Travis was basically a hosted instance of Hudson — or at least similar enough that his experience might help.

—
Ben

> On Jul 9, 2015, at 7:03 AM, Brandon Romano notifications@github.com wrote:
> 
> Hmm -- I'd prefer to keep it travis if at all possible... Perhaps I can dig into it as well. I notice you've got another branch going and it seems like it's coming from the tests... Perhaps one of us can dig into that deeper.
> 
> —
> Reply to this email directly or view it on GitHub https://github.com/BrandonRomano/wrecker/pull/33#issuecomment-119954752.

BrandonRomano commented:
Oh gotcha.  Yeah they are two different things, but if you think someone else's expertise can help, I'm open to them coming and checking it out (this is an open source project for a reason :stuck_out_tongue_winking_eye:)
